encoding/thrift/benchmark_test.go

```diff
@@ -0,0 +1,49 @@
+package thrift
+
+import (
+	"bytes"
+	"context"
+	"testing"
+	"time"
+
+	"go.uber.org/yarpc/api/transport"
+	"go.uber.org/yarpc/api/transport/transporttest"
+
+	"github.com/golang/mock/gomock"
+	"go.uber.org/thriftrw/wire"
+)
+
+func BenchmarkThrift(b *testing.B) {
+	mockCtrl := gomock.NewController(b)
+	defer mockCtrl.Finish()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	proto := NewMockProtocol(mockCtrl)
+	proto.EXPECT().DecodeEnveloped(gomock.Any()).Return(wire.Envelope{
```

Comment:
you are benchmarking the generated mock code at this point. Maybe you only need to call encode/decode in the loop.